crates/sui-swarm-config/src/node_config_builder.rs

```diff
@@ -0,0 +1,299 @@
+                        .udp_multiaddr_to_listen_address()
+                        .unwrap()
+                }),
+                external_address: Some(validator_config.p2p_address),
+                seed_peers,
+                ..Default::default()
+            }
+        };
+
+        let localhost = local_ip_utils::localhost_for_testing();
+        let json_rpc_address = self.rpc_addr.unwrap_or_else(|| {
+            let rpc_port = self
+                .rpc_port
+                .unwrap_or_else(|| local_ip_utils::get_available_port(&ip));
+            format!("{}:{}", ip, rpc_port).parse().unwrap()
+        });
+
+        NodeConfig {
+            protocol_key_pair: AuthorityKeyPairWithPath::new(validator_config.key_pair),
+            account_key_pair: KeyPairWithPath::new(validator_config.account_key_pair),
+            worker_key_pair: KeyPairWithPath::new(SuiKeyPair::Ed25519(
+                validator_config.worker_key_pair,
+            )),
+            network_key_pair: KeyPairWithPath::new(SuiKeyPair::Ed25519(
+                validator_config.network_key_pair,
+            )),
+
+            db_path,
+            network_address: validator_config.network_address,
+            metrics_address: local_ip_utils::new_local_tcp_socket_for_testing(),
+            admin_interface_port: local_ip_utils::get_available_port(&localhost),
+            json_rpc_address,
+            consensus_config: None,
+            enable_event_processing: true, // This is unused.
+            enable_index_processing: default_enable_index_processing(),
+            genesis: sui_config::node::Genesis::new(network_config.genesis.clone()),
+            grpc_load_shed: None,
+            grpc_concurrency_limit: None,
+            p2p_config,
+            authority_store_pruning_config: AuthorityStorePruningConfig::fullnode_config(),
+            end_of_epoch_broadcast_channel_capacity:
+                default_end_of_epoch_broadcast_channel_capacity(),
+            checkpoint_executor_config: Default::default(),
+            metrics: None,
+            supported_protocol_versions: self.supported_protocol_versions,
+            db_checkpoint_config: self.db_checkpoint_config.unwrap_or_default(),
+            indirect_objects_threshold: usize::MAX,
+            expensive_safety_check_config: self
+                .expensive_safety_check_config
+                .unwrap_or_else(ExpensiveSafetyCheckConfig::new_enable_all),
+            name_service_package_address: None,
+            name_service_registry_id: None,
+            name_service_reverse_registry_id: None,
+            transaction_deny_config: Default::default(),
+            certificate_deny_config: Default::default(),
+            state_debug_dump_config: Default::default(),
+        }
+    }
+}
+
+fn get_key_path(key_pair: &AuthorityKeyPair) -> String {
+    let public_key: AuthorityPublicKeyBytes = key_pair.public().into();
+    let mut key_path = Hex::encode(public_key);
+    key_path.truncate(12);
```

Comment:
Nit: make this 12 a constant? or comment what it is?

Reply:
I copied this from old code. It's just to name the directory name of the validator using validator's public key, but make it shorter